th2_data_services/provider/v5/command.py

```diff
@@ -29,3 +32,33 @@ class IGRPCProvider5Command(IGRPCProviderCommand):
     @abstractmethod
     def handle(self, data_source: GRPCProvider5DataSource):
         pass
+
+
+class IGRPCProvider5EventCommand(IGRPCProvider5Command):
+    @abstractmethod
+    def handle(self, data_source: GRPCProvider5DataSource):
+        pass
+
+    @staticmethod
+    def _decode_event(event: EventData) -> dict:
+        new_event = MessageToDict(event, including_default_value_fields=True)
+        try:
+            new_event["body"] = json.loads(event.body)
+        except (KeyError, json.JSONDecodeError):
+            return new_event
+        return new_event
+
+
+class IGRPCProvider5MessageCommand(IGRPCProvider5Command):
+    @abstractmethod
+    def handle(self, data_source: GRPCProvider5DataSource):
+        pass
+
+    @staticmethod
+    def _decode_message(message: MessageData) -> dict:
+        new_message = MessageToDict(message, including_default_value_fields=True)
+        try:
+            new_message["body"] = json.loads(message.body)
+        except (KeyError, json.JSONDecodeError):
+            return new_message
+        return new_message
```

Comment:
I think it's better to use an external decoder class and use it in commands.
A decoder class can be created in Command constructor. 
It's good point to add `update_decoder(Decoder_obj)` method

Reply:
All Decoder_obj should have the same interface.

Reply:
Also, it looks like an sse_adapter. I think we should change sse_adapter to the same idea.